pkg/parse/run.go

```diff
@@ -179,6 +185,26 @@ func Run(ctx context.Context, p Parser) {
 			statusUpdateTimer.Reset(opts.statusUpdatePeriod) // Schedule status update attempt
 			// we should not reset retryTimer under this `case` since it is not aware of the
 			// state of backoff retry.
+
+		// Execute the entire parse-apply-watch loop for a namespace event.
+		case <-nsEventTimer.C:
+			if nsControllerState == nil {
+				// If the Namespace Controller is not running, stop the timer without
+				// closing the channel.
+				nsEventTimer.Stop()
+				continue
+			}
+			if nsControllerState.IsSyncPending() {
+				klog.Infof("A new sync is triggered by a Namespace event")
+				// Reset the cache partially to make sure all the steps of a parse-apply-watch loop will run.
+				// The cached sourceState will not be reset to avoid reading all the source files unnecessarily.
+				// The cached needToRetry will not be reset to avoid resetting the backoff retries.
```

Comment:
Does this mean that the `run` here will just return and skip applying if we're in backoff?
I can't remember how backoff works.
I think I once expected it to be implemented here with these timers but it wasn't.

Reply:
The backoff is only for retry events. The run function doesn't check  the backoff.